rust-toolchain

```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "1.78.0"
+channel = "1.79.0"
```

Comment:
This alone should be a separate PR, let's use the same version as polkadot-sdk has in the stable2407 branch

Reply:
Which version is used in polkadot-sdk? I cannot find a `rust-toolchain` file :/

I bumped to this version to take advantage of [this new feature](https://blog.rust-lang.org/2024/06/13/Rust-1.79.0.html#bounds-in-associated-type-position), which make trait aliases way more powerful as they can now imply associated types bounds (no need to repeat long where clauses everywhere).

Reply:
You can find a way to find the rust version here: https://github.com/paritytech/polkadot-sdk/issues/5335

Reply:
They seem to target 1.77 which doesn't have the feature above. Not using this feature will propagate big where clauses everywhere there is a `RuntimeApi` :/ It should be fine to use a different stable version of Rust than the polkadot repo as long as we all use the same (thanks to `rust-toolchain` file)

Reply:
That's quite old...

I see they are updating directly to 1.81: https://github.com/paritytech/polkadot-sdk/pull/5676

So you can choose 1.78 or 1.81, but let's do it in a separate PR, and let's merge it after the starlight branch is created (I don't want this to break the release process).

Reply:
https://github.com/moondance-labs/tanssi/pull/691